hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/FilterVisitor.java

```diff
@@ -85,7 +90,7 @@ public void visit(GraphTraversal<?, ?> query, Related related, QueryTranslationS
                 if (null != related.getRelationshipName()) {
                     state.setInEdges(true);
                     state.setComingFrom(Direction.IN);
-                    query.inE(related.getRelationshipName());
+                    query.inE(related.getRelationshipName()).discriminate(discriminator);
```

Comment:
no, in here `discriminate()` is very similar to `has()`. It only applies to the current position in the traversal.